operator/text/v0/chunk_text.go

```diff
@@ -0,0 +1,137 @@
+	// TODO: Add SecondSplitter, which is to set the details about how to chunk the paragraphs in Markdown format.
+	// https://pkg.go.dev/github.com/tmc/langchaingo@v0.1.10/textsplitter#MarkdownTextSplitter
+	// secondSplitter textsplitter.TextSplitter
+}
+
+type ChunkTextOutput struct {
+	ChunkNum   int         `json:"chunk_num"`
+	TextChunks []TextChunk `json:"text_chunks"`
+	TokenCount int         `json:"token_count,omitempty"`
+}
+
+type TextChunk struct {
+	Text          string `json:"text"`
+	StartPosition int    `json:"start_position,omitempty"`
+	EndPosition   int    `json:"end_position,omitempty"`
+}
+
+func (s *Setting) SetDefault() {
+	if s.ChunkSize == 0 {
+		s.ChunkSize = 512
+	}
+	if s.ChunkOverlap == 0 {
+		s.ChunkOverlap = 100
+	}
+	if s.ModelName == "" {
+		s.ModelName = "gpt-3.5-turbo"
+	}
+	if s.EncodingName == "" {
+		s.EncodingName = "cl100k_base"
+	}
+	if s.AllowedSpecial == nil {
+		s.AllowedSpecial = []string{}
+	}
+	if s.DisallowedSpecial == nil {
+		s.DisallowedSpecial = []string{"all"}
+	}
+	if s.Separators == nil {
+		s.Separators = []string{"\n\n", "\n", " ", ""}
+	}
+}
+
+func chunkText(input ChunkTextInput) (ChunkTextOutput, error) {
+	var split textsplitter.TextSplitter
+	setting := input.Strategy.Setting
+	setting.SetDefault()
+	switch setting.ChunkMethod {
+	case "Token":
+		
+		if setting.ChunkOverlap >= setting.ChunkSize {
+			err := fmt.Errorf("ChunkOverlap must be less than ChunkSize when using Token method")
+			return ChunkTextOutput{}, err
+		}
+
+		split = textsplitter.NewTokenSplitter(
+			textsplitter.WithChunkSize(setting.ChunkSize),
+			textsplitter.WithChunkOverlap(setting.ChunkOverlap),
+			textsplitter.WithModelName(setting.ModelName),
+			textsplitter.WithEncodingName(setting.EncodingName),
+			textsplitter.WithAllowedSpecial(setting.AllowedSpecial),
+			textsplitter.WithDisallowedSpecial(setting.DisallowedSpecial),
+		)
+	case "Markdown":
+		split = textsplitter.NewMarkdownTextSplitter(
+			textsplitter.WithChunkSize(setting.ChunkSize),
+			textsplitter.WithChunkOverlap(setting.ChunkOverlap),
+			textsplitter.WithCodeBlocks(setting.CodeBlocks),
+			textsplitter.WithReferenceLinks(setting.ReferenceLinks),
+		)
+	case "Recursive":
+		split = textsplitter.NewRecursiveCharacter(
+			textsplitter.WithSeparators(setting.Separators),
+			textsplitter.WithChunkSize(setting.ChunkSize),
+			textsplitter.WithChunkOverlap(setting.ChunkOverlap),
+			textsplitter.WithKeepSeparator(setting.KeepSeparator),
+		)
+	}
+
+	chunks, err := split.SplitText(input.Text)
+	if err != nil {
+		return ChunkTextOutput{}, err
+	}
+
+	output := ChunkTextOutput{
+		ChunkNum: len(chunks),
+	}
+
+	if setting.ChunkMethod == "Token" {
+		tkm, err := tiktoken.EncodingForModel(setting.ModelName)
+		if err != nil {
+			return ChunkTextOutput{}, err
+		}
+		token := tkm.Encode(input.Text, setting.AllowedSpecial, setting.DisallowedSpecial)
+		output.TokenCount = len(token)
+	}
```

Comment:
This can go in the `switch` statement above

Reply:
As mentioned in a previous comment, I'd avoid external calls in this document. You can set an offline BPE loader by using https://github.com/pkoukk/tiktoken-go-loader

Reply:
[Replied here](https://github.com/instill-ai/component/pull/139#discussion_r1618440907)

Reply:
As mentioned, I'm fine keeping this package and updating Pipeline to use it, too. But if we do so, I'd add the offline loader to avoid external calls.